src/v/pandaproxy/util.h

```diff
@@ -23,20 +24,17 @@ namespace pandaproxy {
 /// wait for it to finish. Concurrent invocatons will also wait.
 ///
 /// On success, all waiters will be allowed to continue. Successive invocations
-/// of one_shot::operator()() will return ss::now().
+/// of one_shot::operator()() will return a ready future.
 ///
 /// If func fails, waiters will receive the error, and one_shot will be reset.
 /// Successive calls to operator()() will restart the process.
 class one_shot {
-    enum class state { empty, started, available };
-    using futurator = ss::futurize<ssx::semaphore_units>;
-
 public:
     explicit one_shot(ss::noncopyable_function<ss::future<>()> func)
       : _func{std::move(func)} {}
-    futurator::type operator()() {
+    ss::future<> operator()() {
         if (likely(_started_sem.available_units() != 0)) {
-            return ss::get_units(_started_sem, 1);
+            return ss::get_units(_started_sem, 1).discard_result();
```

Comment:
```suggestion
            return _started_sem.wait();
```

Reply:
Uh actually this doesn't release the count back so we shouldn't have switched this to wait (this just never gives back the units). Probably not a big deal because we make this max size_t, but we should put the counter back. I've opened https://github.com/redpanda-data/redpanda/pull/15278 to fix